src/app/task.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { Tasks } from './tasks';
+import { tasks } from './mock-tasks';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskService {
+  private tasks: Tasks[] = tasks;
+  getTasks(): Observable<Tasks[]> {
+    return of(tasks);
+  }
+  addTask(newTask: Tasks): void {
+    newTask.id = this.generateUniqueId();
+       this.tasks.push(newTask)
+    // this.tasks = [...this.tasks, newTask]
```

Comment:
try to remove the commented code before creating PR and if you need such code then add a TODO